contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,177 @@
+
+>Angular started as a side project. Back in 2009, Miško Hevery and Adam Abrons released a project that would help developers build web applications using simple HTML tags. The name “Angular” comes from the angle brackets, or < >, that surround all HTML tags.
+>
+<sub>Miško described the idea behind the framework in an <a href="https://www.infoworld.com/article/2612801/javascript/whats-so-special-about-googles-angularjs.html">interview done in 2013</a></sub>
+
+Angular is a complex framework and can be challenging for new developers to understand.
+However, given below are some noteworthy features of Angular (adapted from [here](https://hackr.io/blog/why-should-you-learn-angular-in-2018)):
+
+1. **Angular supports Cross Platform Development.** Apart from web applications, the more recent versions of Angular can be used to build <tooltip content="a smartphone application that is coded in a specific programming language, such as Swift for iOS or Java for Android operating systems">native mobile applications</tooltip> and Desktop applications.
+ With Angular 5 and above, progressive web applications can also be developed. These applications have high performance and can work offline as well.
+
+2. **Angular uses TypeScript.** Angular applications are built using the TypeScript language, a superset of JavaScript. TypeScript provides <tooltip content="Static typing involves type checking during compile-time as opposed to run-time. Types are associated with variables not values.">static typing</tooltip>. This helps the compiler show warnings about any potential errors in the code, before the app runs. Another advantage is that TypeScript provides code completion using <tooltip content="IntelliSense provides active hints as code is added">IntelliSense.</tooltip>
+
+    Below is an example of static typing using TypeScript.
+    
+    ```typescript
+    // TypeScript
+    let name: string; // <-- static typed!
+    ```
+    
+    ```javascript
+    // JavaScript (ES6)
+    let name;
+    ```
+    <box type="info"> TypeScript was created and is maintained by Microsoft. It has been <a href="https://developer.telerik.com/featured/the-rise-of-typescript/">growing in popularity</a> over the last few years making it reliable to use.</box>
+                                                  
+3. **Angular is backed by Google.** Google announced Long-Term Support (LTS) for Angular back in 2017. This means Google plans to further scale up the Angular ecosystem.
+
+### What does an Angular app look like?
+
+Let’s dive into some code, with a hello world example. The code is applicable for versions Angular 2 and above.
+
+```html
+<!doctype html>
+<html>
+  <head>
+    <!-- The various &lt;script&gt; tags needed to load Angular -->
+    <!-- For a full list, see https://angular.io/docs/ts/latest/quickstart.html -->
+  </head>
+  <body>
+    <my-app></my-app>
+  </body>
+</html>
+```
```

Comment:
I noticed Angular website has its own helloworld example. Consider reusing/adapting that over creating your own.

Reply:
Just an observation. I looked at the example on the official website. It is not a common hello world example. Instead, it is meant to explain quite a few concepts in Angular as opposed to giving the reader a brief overview of what "An angular app looks like" IMO it may be a lot of any newcomer to take in. What do you think?